subsys/bluetooth/mesh/sensor_srv.c

```diff
@@ -824,20 +852,25 @@ static int update_handler(struct bt_mesh_model *model)
 	       bt_mesh_model_pub_period_get(model), period_div,
 	       srv->pub.fast_period ? "fast" : "normal");
 
+	/** Reset the publication divisor and the fast period divisor flag to get the actual
+	 * publication period.
+	 */
 	srv->pub.period_div = 0;
 	srv->pub.fast_period = 0;
 
 	uint32_t base_period = bt_mesh_model_pub_period_get(model);
 
+	srv->pub.fast_period = true;
```

Comment:
Two lines above this is set to "0" ... :)

Reply:
Yes, to make `bt_mesh_model_pub_period_get` return actual publication period without dividing in to fast cadence period divisor.

Reply:
This looks really awkward. Like hacking the `bt_mesh_model_pub_period_get()` API. It will be worth adding a comment at least just before setting fast_period to 0 (or rather "false").

Reply:
But the comment is already there:
https://github.com/nrfconnect/sdk-nrf/blob/6c35a8afd9b5fec5f4fef7e6f66a6215f7c497de/subsys/bluetooth/mesh/sensor_srv.c#L855-L857